File: backend/models/ratingModel.js
// ratingModel.js
const mongoose = require("mongoose");

const ratingSchema = new mongoose.Schema({
  doctorId: {
    type: String,
    required: true,
  },
  
  userId: { type: String, required: true },
  rating: { type: Number, min: 1, max: 5, required: true },
  message: { type: String, max: 500 },
});


const Rating = mongoose.model("Rating", ratingSchema);

module.exports = Rating;

File: Frontend/tenaye/lib/Infrastructure/local/mock_model/rating_model.dart
class Rating {
  final String? id; // New field: rating ID
  final String? doctorId;
  final String? userId;
  final String message;
  final double rating;

  Rating({
    this.id,
    this.doctorId,
    this.userId,
    required this.message,
    required this.rating,
  });

  factory Rating.fromMap(Map<String, dynamic> map) {
    return Rating(
      id: map['id'],
      doctorId: map['doctorId'],
      userId: map['userId'],
      message: map['message'],
      rating: map['rating'].toDouble(),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'doctorId': doctorId,
      'userId': userId,
      'message': message,
      'rating': rating,
    };
  }
}
